btcwallet/btcwallet_test.go

```diff
@@ -0,0 +1,128 @@
+package btcwallet
+
+import (
+	"errors"
+	"sync"
+	"testing"
+)
+
+func TestBtcWallet_Race(t *testing.T) {
+	var startBalance Bitcoin = 20
+	var expected Bitcoin = 50
+	wallet := NewBtcWallet(startBalance)
+	wg := &sync.WaitGroup{}
+
+	for i := 0; i < 3; i++ {
+		wg.Add(1)
+		go func(w *BtcWallet) {
```

Comment:
No need to pass the reference to the wallet, the goroutine will have access to it